opentracing-shim/tests/test_shim.py

```diff
@@ -0,0 +1,386 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+import opentracing
+
+import opentracingshim
+from opentelemetry import trace
+from opentelemetry.sdk.trace import Tracer
+from opentracingshim import util
+
+
+class TestShim(unittest.TestCase):
+    # pylint: disable=too-many-public-methods
+
+    def setUp(self):
+        # TODO: Review setup.
+        self.tracer = trace.tracer()
+        self.shim = opentracingshim.create_tracer(self.tracer)
+
+    @classmethod
+    def setUpClass(cls):
+        """Set preferred tracer implementation only once rather than before
+        every test method.
+        """
+        # TODO: Do we need to call setUpClass() on super()?
+        # Seems to work fine without it.
+        super(TestShim, cls).setUpClass()
+        trace.set_preferred_tracer_implementation(lambda T: Tracer())
+
+    def test_shim_type(self):
+        # Verify shim is an OpenTracing tracer.
+        self.assertIsInstance(self.shim, opentracing.Tracer)
+
+    def test_start_active_span(self):
+        """Test span creation and activation using `start_active_span()`."""
+
+        with self.shim.start_active_span("TestSpan") as scope:
+            # Verify correct type of Scope and Span objects.
+            self.assertIsInstance(scope, opentracing.Scope)
+            self.assertIsInstance(scope.span, opentracing.Span)
+
+            # Verify span is started.
+            self.assertIsNotNone(scope.span.unwrap().start_time)
+
+            # Verify span is active.
+            self.assertEqual(self.shim.active_span.context, scope.span.context)
+            # TODO: We can't check for equality of self.shim.active_span and
```

Comment:
You could also override `__hash__` and `__equal__` to be based on the wrapped OTel object. But I don't know if that makes sense.

Reply:
I'd rather solve this using your [suggestion](https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274). This would also be more efficient than re-creating wrapper objects.